File: 002-多线程测试/002-多线程测试/ViewController.m
//
//  ViewController.m
//  002-多线程测试
//
//  Created by 马遥 on 2018/1/11.
//  Copyright © 2018年 马遥. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    NSLog(@"1");
    dispatch_queue_t queue = dispatch_queue_create("chuanxingduilie", NULL);
    dispatch_async(queue, ^{
        NSLog(@"2");
        dispatch_async(queue, ^{
            NSLog(@"3");
        });
        NSLog(@"4");
    });
    NSLog(@"5");
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: swift_test/swift_test/ViewController.swift
//
//  ViewController.swift
//  swift_test
//
//  Created by 马遥 on 2018/1/9.
//  Copyright © 2018年 马遥. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        var customersInLine = ["Chris", "Alex", "Ewa", "Barry", "Daniella"]
        print(customersInLine.count)
        // 打印出 "5"
        let customerProvider = { customersInLine.remove(at: 0) }
        print(customersInLine.count)
        // 打印出 "5"
        print("Now serving \(customerProvider())!") // Prints "Now serving Chris!"
        print(customersInLine.count)
        // 打印出 "4"
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

